src/prommis/uky/costing/tests/test_ree_costing.py

```diff
@@ -1949,7 +1986,314 @@ def test_REE_watertap_costing(self, model, solver):
         )
 
         assert model.fs.costing.total_fixed_OM_cost.value == pytest.approx(
-            11.50202, rel=1e-4
+            11.68508, rel=1e-4
+        )
+
+        assert pyo.value(
+            pyunits.convert(
+                model.fs_membrane.nfzounit.costing.variable_operating_cost,
+                to_units=CE_index_units / pyunits.year,
+            )
+        ) == pytest.approx(0.410881, rel=1e-4)
+
+        assert model.fs.costing.watertap_variable_costs.value == pytest.approx(
+            0.410881, rel=1e-4
+        )
+
+        assert model.fs.costing.total_variable_OM_cost[0].value == pytest.approx(
+            533.42952, rel=1e-4
+        )
+
+
+class TestCustomCosting(object):
+    @pytest.fixture(scope="class")
+    def model(self):
+        model = base_model()
+
+        # create model
+        model.fs.custom_vessel = UnitModelBlock()
+        model.fs.custom_vessel.volume = pyo.Var(initialize=5000, units=pyunits.m**3)
+        model.fs.custom_vessel.volume.fix()
+        model.fs.custom_vessel.water_injection_rate = pyo.Var(
+            initialize=0.1, units=pyunits.m**3 / pyunits.s
+        )
+        model.fs.custom_vessel.water_injection_rate.fix()
+
+        # add costing
+        model.fs.custom_vessel.costing = UnitModelCostingBlock(
+            flowsheet_costing_block=model.fs.costing,
+            costing_method=CustomCostingData.cost_custom_vessel,
+            costing_method_arguments={
+                "volume_per_unit": model.fs.custom_vessel.volume,
+                "material": "carbonsteel",
+                "water_injection_rate_per_unit": model.fs.custom_vessel.water_injection_rate,
+                "number_of_units": 3,
+                "CE_index_year": "2022",
+            },
+        )
+
+        return model
+
+    @pytest.mark.component
+    def test_REE_custom_costing(self, model):
+        # full smoke test with all components, O&M costs, and extra costs included
+        CE_index_year = "UKy_2019"
+
+        CE_index_units = getattr(
+            pyunits, "MUSD_" + CE_index_year
+        )  # millions of USD, for base year
+
+        # add plant-level cost constraints
+
+        model.fs.feed_input = pyo.Var(initialize=500, units=pyunits.ton / pyunits.hr)
+        model.fs.feed_grade = pyo.Var(initialize=356.64, units=pyunits.ppm)
+
+        hours_per_shift = 8
+        shifts_per_day = 3
+        operating_days_per_year = 336
+
+        # for convenience
+        model.fs.annual_operating_hours = pyo.Param(
+            initialize=hours_per_shift * shifts_per_day * operating_days_per_year,
+            mutable=False,
+            units=pyunits.hours / pyunits.a,
```

Comment:
If this will be used as an example for students or new costing users, using `pyunits.yr` instead of `pyunits.a` may be more helpful (this was not immediately obvious to me).